sdcm/sla/libs/sla_utils.py

```diff
@@ -0,0 +1,285 @@
+#!/usr/bin/env python
+import logging
+
+from sdcm.sct_events import Severity
+from sdcm.sct_events.workload_prioritisation import WorkloadPrioritisationEvent
+from test_lib.sla import Role
+
+LOGGER = logging.getLogger(__name__)
+
+
+class SchedulerRuntimeUnexpectedValue(Exception):
+    pass
+
+
+class SlaUtils:
+    STRESS_READ_CMD = 'cassandra-stress read cl=QUORUM duration={duration} -mode cql3 native user={user} ' \
+        'password={password} -rate threads={threads} -pop {pop}'
+    WORKLOAD_LATENCY = 'latency'
+    WORKLOAD_THROUGHPUT = 'throughput'
+    CACHE_ONLY_LOAD = 'cache_only'
+    DISK_ONLY_LOAD = 'disk_only'
+    MIXED_LOAD = 'mixed'
+    MIN_CPU_UTILIZATION = 97
+    VALID_DEVIATION_PRC = 10
+
+    @staticmethod
+    # pylint: disable=too-many-arguments, too-many-locals
+    def define_read_cassandra_stress_command(role: Role,
+                                             load_type: str,
+                                             c_s_workload_type: str,
+                                             threads: int,
+                                             stress_duration_min: int,
+                                             max_rows_for_read: int = None,
+                                             stress_command: str = STRESS_READ_CMD,
+                                             throttle: int = 20000,
+                                             num_of_partitions=50000000,
+                                             **kwargs):
+        """
+        :param role: Role object
+        :param load_type: cache_only/disk_only/mixed
+        :param c_s_workload_type: latency: with ops restriction - using throttle
+                                or
+                              throughput: no restriction
+        """
+
+        def latency():
+            return '%d throttle=%d/s' % (threads, throttle)
+
+        def throughput():  # pylint: disable=unused-variable
+            return threads
+
+        def cache_only(max_rows_for_read):  # pylint: disable=unused-variable
+            if not max_rows_for_read:
+                max_rows_for_read = int(num_of_partitions * 0.3)
+            return 'seq=1..%d' % max_rows_for_read
+
+        # Read from cache and disk
+        def mixed(max_rows_for_read):  # pylint: disable=unused-variable
+            if not max_rows_for_read:
+                max_rows_for_read = num_of_partitions
+            return "'dist=gauss(1..%d, %d, %d)'" % (max_rows_for_read,
+                                                    int(max_rows_for_read / 2),
+                                                    int(max_rows_for_read * 0.05))
```

Comment:
This is not accurate that this is what defines the cache or disk.
This defined the distribution we use.
What controls the cache or disk is the number of partitions.
If we don't have enough partitions, both distributions will be from cache, and if we have too many, both will involve the disk.

Reply:
Added explanation.
Actually this code is from `sla_per_user_system_test.py`: https://github.com/scylladb/scylla-cluster-tests/blob/66491ebf531b1a2d95959e4972bd61574660ee79/sla_per_user_system_test.py#L305
and used in the feature test. I copied this part to the new SLA common module because I use it also in longevity now. 
I plan to remove it from `sla_per_user_system_test.py` but not in this PR.
We decided together about this stress definition 3 years ago and wanted that data read distribution was like this.